src/python/pants/backend/go/subsystems/golang.py

```diff
@@ -93,6 +100,76 @@ class GolangSubsystem(Subsystem):
         advanced=True,
     )
 
+    cgo_enabled = BoolOption(
+        "--cgo-enabled",
+        default=False,
+        help=softwrap(
+            """\
+            Enable Cgo, which allows calling C code from Go.
+
+            See https://go.dev/blog/cgo and https://pkg.go.dev/cmd/cgo for additional information.
+            """
+        ),
+    )
+
+    _cgo_tool_search_paths = StrListOption(
+        default=["<PATH>"],
+        help=softwrap(
+            """
+            A list of paths to search for tools needed by CGo (e.g., gcc, g++).
+
+            Specify absolute paths to directories with tools needed by CGo , e.g. `/usr/bin`.
+            Earlier entries will be searched first.
+
+            The following special strings are supported:
+
+              * `<PATH>`, the contents of the PATH environment variable
+            """
+        ),
+    )
+
+    cgo_gcc_binary_name = StrOption(
+        default="gcc",
+        advanced=True,
+        help="Name of the tool to use to compile C code included via CGo in a Go package.",
+    )
+
+    cgo_gxx_binary_name = StrOption(
+        default="g++",
+        advanced=True,
+        help="Name of the tool to use to compile C++ code included via CGo in a Go package.",
+    )
+
+    cgo_fortran_binary_name = StrOption(
+        default="gfortran",
+        advanced=True,
+        help="Name of the tool to use to compile fortran code included via CGo in a Go package.",
+    )
+
+    cgo_default_cflags = StrListOption(
+        default=lambda _: list(_DEFAULT_COMPILER_FLAGS),
+        advanced=True,
+        help="Default options used for CFLAGS when compiling cgo code. Equivalent to CGO_CFLAGS passed to `go`.",
+    )
+
+    cgo_default_cxxflags = StrListOption(
+        default=lambda _: list(_DEFAULT_COMPILER_FLAGS),
+        advanced=True,
+        help="Default options used for CXXFLAGS when compiling cgo code. Equivalent to CGO_CXXFLAGS passed to `go`.",
+    )
+
+    cgo_default_fflags = StrListOption(
+        default=lambda _: list(_DEFAULT_COMPILER_FLAGS),
+        advanced=True,
+        help="Default options used for FFLAGS when compiling cgo code. Equivalent to CGO_FFLAGS passed to `go`.",
+    )
+
+    cgo_default_ldflags = StrListOption(
```

Comment:
Using the term `ldflags` here is done intentionally to line up with CGO's args, correct?

In lieu of something like `link_flags` or `link_options` - this is to be clear that this option directly passes into `CGO_LDFLAGS`?

Reply:
Yes, I chose `ldflags` to match with its use with `CGO_LDFLAGS`. I assume go developers will be familiar with the `go` terminology here.

Reply:
For reference, [Bazel rules_go naming for fields on targets related to compile/link options](https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/rules.md#go_binary): 
- `copts` == CGO_CFLAGS
- `cppopts` == CGO_CPPFLAGS
- `cxxopts` == CGO_CXXFLAGS
- `clinkopts` == CGO_LDFLAGS

So there is some precedent in not using `cflags` / `ldflags` naming. Although I'm not particularly in favor of `clinkopts` as any better than `ldflags`. Going verbose here and having it be similar to the `cc` backend's naming choices will lead to a more consistent experience for Pants users.

Reply:
Tangentially, I note that the `cc` backend is using "cpp" to refer to C++ while Cgo uses "cpp" to refer to the C/C++ preprocessor and "cxx" to refer to C++. (Go will prepend the "cpp" flags to both "cflags" and "cxxflags.")

@sureshjoshi : Thoughts on cpp versus cxx naming?

Reply:
I personally rather dislike how Bazel names options (in the CC backend especially - `copts`, `hdrs`, etc). Using word short-forms, no spacing, ugh - feels very 90's or 00's coding.

Anyways, I agree that keeping `cgo` in line with `cc` will be useful. 

I'm finishing up some unit tests, then I'll push a preliminary version for a biiiiig bikeshed on naming. I'm using CPP, only because what's what I've always used for CPlusPlus. 

I'm guessing Bazel is using more of a Makefile convention (https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html).

CMake uses CXX as well - https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html

I've got no opposition to changing and aligning with other tooling, I ran with CPP just so I wouldn't waste time on naming when there was a whole backend to write, and the CXX vs CPP decision would be bigger than just my opinion anyways. 

In general, verbosely following existing conventions are my preference - plain English reading where possible.

Reply:
🤦🏽 
Completely forgot that `gcc` uses CPP as C pre-processor in docs.

I almost never touch those options directly, so I don't even think about it anymore. Probably a good reason to use `cxx` and never use `cpp`... Even though all my C++ files are named `.cpp`